src/codemodder/codemods/imported_call_modifier.py

```diff
@@ -0,0 +1,90 @@
+import abc
+from typing import Sequence
+
+import libcst as cst
+from libcst import matchers
+from libcst.codemod import CodemodContext, VisitorBasedCodemodCommand
+from libcst.metadata import PositionProvider
+
+from codemodder.change import Change
+from codemodder.codemods.utils_mixin import NameResolutionMixin
+from codemodder.file_context import FileContext
+
+
+class ImportedCallModifier(
+    VisitorBasedCodemodCommand, NameResolutionMixin, metaclass=abc.ABCMeta
+):
+    METADATA_DEPENDENCIES = (PositionProvider,)
+
+    def __init__(
+        self,
+        codemod_context: CodemodContext,
+        file_context: FileContext,
+        matching_functions: dict[str, str],
+        change_description: str,
+    ):
+        super().__init__(codemod_context)
+        self.line_exclude = file_context.line_exclude
+        self.line_include = file_context.line_include
+        self.matching_functions = matching_functions
+        self._matching_function_names = list(matching_functions.keys())
+        self.change_description = change_description
+        self.changes_in_file: list[Change] = []
+
+    def updated_args(self, original_args: Sequence[cst.Arg]):
+        return original_args
+
+    @abc.abstractmethod
+    def update_attribute(self, true_name, original_node, updated_node, new_args):
+        """Callback to modify tree when the detected call is of the form a.call()"""
+
+    @abc.abstractmethod
+    def update_simple_name(self, true_name, original_node, updated_node, new_args):
+        """Callback to modify tree when the detected call is of the form call()"""
+
+    def leave_Call(self, original_node: cst.Call, updated_node: cst.Call):
+        pos_to_match = self.node_position(original_node)
+        line_number = pos_to_match.start.line
+        if self.filter_by_path_includes_or_excludes(pos_to_match):
+            true_name = self.find_base_name(original_node.func)
+            if (
+                self._is_direct_call_from_imported_module(original_node)
+                and true_name in self._matching_function_names
+            ):
+                self.changes_in_file.append(
+                    Change(str(line_number), self.change_description).to_json()
+                )
+
+                new_args = self.updated_args(updated_node.args)
+
+                # has a prefix, e.g. a.call() -> a.new_call()
+                if matchers.matches(original_node.func, matchers.Attribute()):
+                    return self.update_attribute(
+                        true_name, original_node, updated_node, new_args
+                    )
+
+                # it is a simple name, e.g. call() -> module.new_call()
+                return self.update_simple_name(
+                    true_name, original_node, updated_node, new_args
+                )
```

Comment:
Just a small clarification om this bit of code here.
Within the context of https codemod this comment and logic was accurate. But this may not be general enough.
Consider:
```python
from lib import call
call()
```
We have the option to simply change the call...
```python
from lib import call
import lib2
lib2.call2()
```
or changing the call and import:
```python
from lib2 import call2
call2()
```
The second has an advantage of not having a possibly useless import (not much of a problem since we have a codemod to amend that). Then again, then reason why I've favored the first one for the https codemod was because the change became more explicit.

The second transformation is a bit harder to do but within the realm of possibility. Just tell me if you want me to support that.

Reply:
@andrecsilva that makes sense. It might be good to revisit at some point but right now I like that the updated call becomes completely explicit.